### DISCUSSION TITLE
Input from NI DAQ to Bonsai

### DISCUSSION THREAD

**Discussion by xingberlin** [bonsai-rx/bonsai#1420] [Q&A]
We want to send analog input and digital input from the NI DAQ (National Instrument connected with Neuropixel) to Bonsai, so they can be syncronized with another signal sent to Bonsai from a camera. 

We installed DAQmx package and add Source-DAQmx-analog input or -digital input into the workflow, but Bonsai requires at least one input for the 'analog input' to work. I tired to type in channel number but it didn't help. 

How does this works please? Unfortunately, I did not find tutorial or detailed instruction of this online.

**Comment by bruno-f-cruz** [comment]
Hi, 

Right click the node > view help or F1)
It should take you [here](https://bonsai-rx.org/daqmx/)

**Reply by sunreetab** [reply]
Is the Bonsai-DAQmx documentation ready?

**Reply by bruno-f-cruz** [reply]
Can you clarify? Not sure I understand the question

**Reply by sunreetab** [reply]
The linked page seems to be under development
![Screenshot 2024-07-19 134203](https://github.com/user-attachments/assets/a3ba1e02-61f1-4dd1-979a-5fb3bd6da4ff)

**Reply by bruno-f-cruz** [reply]
Examples and subject to change, but the current examples are up to date and should work :). If any of the examples is not clear let us know !

**Reply by sunreetab** [reply]
Hi, I'm trying to read a digital input (from a capacitive sensor) into bonsai through a USB NI device. When I use the timer -> digital input example the app crashes every time. Is there a resource somewhere to get started reading inputs via daqmx, possibly with usable examples?
Also, does it matter if the input I'm trying to read is the output of an arduino uno? 
Thanks for your help in advance.

**Reply by bruno-f-cruz** [reply]
What happens if you run without the timer? It may be possible that your hardware does not support the trigger mode overload of the operator.

**Reply by sunreetab** [reply]
Thanks for your reply.

Without a timer the process won't start at all, as an error appears with a sampling type/buffer size problem (will share a screenshot tomorrow). It is concerning that the hardware may not support the necessary function. (a) Is there a way to check it for the NI DAQ device I am using? (b) Is there a workaround with different data collection, timing and triggering settings? It is highly unlikely we can replace our DAQ device.

**Reply by sunreetab** [reply]
![analogInputError_DAQmxBonsai](https://github.com/user-attachments/assets/0982f4fd-bafa-4dba-ab33-d09a243be5fd)

This is the error I see before bonsai crashes trying to read an analog input. I verified using NI MAX that the analog input is read by the DAQ device. This kind of memory access error crops up on all the example analog workflows.

**Reply by bruno-f-cruz** [reply]
What is the version of nimax you have installed? Can you try 19.0?

**Reply by sunreetab** [reply]
NI MAX is the 2024 Q2 release.
Daqmx is the 2024 Q3 release
Do you mean 19.0 for DAQmx or MAX? Thanks!

**Reply by sunreetab** [reply]
Is bonsai compatible with only certain DAQ devices? For example is there a reason USB DAQ devices won't work?

**Reply by bruno-f-cruz** [reply]
> NI MAX is the 2024 Q2 release. Daqmx is the 2024 Q3 release Do you mean 19.0 for DAQmx or MAX? Thanks!

https://github.com/orgs/bonsai-rx/discussions/1638
https://github.com/orgs/bonsai-rx/discussions/1885

Please try one of the previously suggested solutions. I think that ni max installs DAQmx as a dependency, so they should be locked to each other if you install ni max.

Bonsai doesn't care about the device you use. The question is whether the device you are using supports the API calls bonsai is making. For instance, if it has buffered digital input capabilities or not
